graphql-java-servlet/src/main/java/graphql/kickstart/servlet/HttpRequestHandlerImpl.java

```diff
@@ -48,37 +51,61 @@ public void handle(HttpServletRequest request, HttpServletResponse response) thr
 
   private void execute(GraphQLInvocationInput invocationInput, HttpServletRequest request,
       HttpServletResponse response) {
-    try {
-      GraphQLQueryResult queryResult = invoke(invocationInput, request, response);
+    if (request.isAsyncSupported()) {
+      AsyncContext asyncContext = request.isAsyncStarted()
+          ? request.getAsyncContext()
+          : request.startAsync(request, response);
+      asyncContext.setTimeout(configuration.getAsyncTimeout());
+      invoke(invocationInput, request, response)
+          .thenAccept(result -> writeResultResponse(result, request, response))
+          .exceptionally(t -> writeErrorResponse(t, response))
+          .thenAccept(aVoid -> asyncContext.complete());
+    } else {
+      try {
+        GraphQLQueryResult result = invoke(invocationInput, request, response).join();
+        writeResultResponse(result, request, response);
+      } catch (Throwable t) {
+        writeErrorResponse(t, response);
+      }
+    }
+  }
 
-      QueryResponseWriter queryResponseWriter = QueryResponseWriter.createWriter(queryResult, configuration.getObjectMapper(),
-          configuration.getSubscriptionTimeout());
+  private void writeResultResponse(GraphQLQueryResult queryResult, HttpServletRequest request,
+      HttpServletResponse response) {
+    QueryResponseWriter queryResponseWriter = QueryResponseWriter.createWriter(queryResult, configuration.getObjectMapper(),
+        configuration.getSubscriptionTimeout());
+    try {
       queryResponseWriter.write(request, response);
-    } catch (Throwable t) {
-      response.setStatus(STATUS_BAD_REQUEST);
-      log.info("Bad GET request: path was not \"/schema.json\" or no query variable named \"query\" given");
-      log.debug("Possibly due to exception: ", t);
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
     }
   }
 
-  private GraphQLQueryResult invoke(GraphQLInvocationInput invocationInput, HttpServletRequest request,
+  private Void writeErrorResponse(Throwable t, HttpServletResponse response) {
+    response.setStatus(STATUS_BAD_REQUEST);
+    log.info("Bad GET request: path was not \"/schema.json\" or no query variable named \"query\" given");
```

Comment:
You can make this `log.info("Bad GET request: path was not \"/schema.json\" or no query variable named \"query\" given", t);` and remove debug line entirely.

Reply:
I copied that from the original code, I guess maybe the intention was to no pollute the INFO logs with stacktraces... But the class already does so in other places anyway, so yeah I can change it.